### ISSUE TITLE
Add `createdBy` and `assignedTo` in task creation

### ISSUE THREAD

**Issue by swarajpure** [Real-Dev-Squad/website-backend#306]
API contract [here](https://github.com/Real-Dev-Squad/website-api-contracts/tree/main/tasks)

While creating a task using `POST /tasks` we need to add two more fields:

`createdBy`
- This is to track who created that particular task.
- This field shouldn't be sent from the frontend, it should be taken from the cookie itself on backend (reasons: I could send `elon-musk` in `createdBy` field from frontend, which is not something which we want 😉 )
- This is going to be the `userId` of the user and not the `username` since `userId` will remain constant for a user, but `username` may change
- Can be accessed via `req.userData.id`

`assignedTo`
- This will track whom the task is assigned to
- This will be coming from frontend as a string.
- Frontend will be sending `username` here in `req.body`, we need to find `userId` corresponding to the given `username` and store that in the DB (there are functions in `utils/users.js` which will help you do this)

It should be clear by now that we will be storing `userId`s everywhere in the DB. There's a catch though, when we are recieving and sending data, it HAS to be `username` -- we can't expect developers to remember each other's `userId`s 😛 

So in short, request and response will contain `username` but the DB will always contain the corresponding `userId`

**Comment by rashid301** [comment]
I will work on it.
